tensorboard/data/server/types.rs

```diff
@@ -89,13 +92,36 @@ impl Borrow<str> for Run {
     }
 }
 
+/// A map defining how many samples per plugin to keep.
+#[derive(Debug)]
+pub struct PluginSamplingHint(pub HashMap<String, usize>);
+
+impl FromStr for PluginSamplingHint {
+    type Err = <usize as FromStr>::Err;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let mut result = HashMap::new();
+        for pair_str in s.split(',') {
+            let pair: Vec<_> = pair_str.split('=').collect();
+            if pair.len() != 2 {
```

Comment:
If I’m reading this right, it looks like `from_str("")` fails to parse.
IMHO it would be nice for that to successfully yield an empty map: it’s
more consistent, you could drop the `if samples_per_plugin:` on the
Python side, and the CLI parser could just set `default_value = ""`\*.
Thoughts?

\* Due to clap-rs/clap#1740, you may also need to explicitly indicate
that empty values are allowed for this argument, due to a misfeature of
Clap 3.x. Cf. [`gsutil.rs`] for an example. Hopefully this is fixed
before 3.0 final is released.

[`gsutil.rs`]: https://github.com/tensorflow/tensorboard/blob/e68b9646ce245749fd61cb28818284290581501f/tensorboard/data/server/gcs/gsutil.rs#L43